controller/setting_controller.go

```diff
@@ -1213,12 +1230,12 @@ func (sc *SettingController) enqueueSettingForBackupTarget(obj interface{}) {
 }
 
 // updateInstanceManagerCPURequest deletes all instance manager pods immediately with the updated CPU request.
-func (sc *SettingController) updateInstanceManagerCPURequest() error {
-	imPodList, err := sc.ds.ListInstanceManagerPods()
+func (sc *SettingController) updateInstanceManagerCPURequest(dataEngine longhorn.DataEngineType) error {
+	imPodList, err := sc.ds.ListInstanceManagerPodsBy("", "", longhorn.InstanceManagerTypeAllInOne, dataEngine)
```

Comment:
For v1, should we query all instance types including engine and replica IM pos in case there could be some old IM pods running? 

```go
ListInstanceManagerPodsBy("", "", dataEngine, longhorn.InstanceManagerTypeAllInOne, InstanceManagerTypeEngine, InstanceManagerTypeReplica)
```

Like

[/controller/kubernetes_secret_controller.go#L214-L224](https://github.com/longhorn/longhorn-manager/blob/98189bfad382f206c0a6ba13e1dbde6283707795/controller/kubernetes_secret_controller.go#L214-L224)
```go
	replicaInstanceManagerPods, err := ks.ds.ListInstanceManagerPodsBy(ks.controllerID, "", longhorn.InstanceManagerTypeReplica, "")
	if err != nil {
		return err
	}
	pods := append(managerPods, replicaInstanceManagerPods...)

	aioInstanceManagerPods, err := ks.ds.ListInstanceManagerPodsBy(ks.controllerID, "", longhorn.InstanceManagerTypeAllInOne, "")
	if err != nil {
		return err
	}
	pods = append(pods, aioInstanceManagerPods...)
```

Reply:
>For v1, should we query all instance types including engine and replica IM pos in case there could be some old IM pods running?

IIUC, 
- If the upgrade is from v1.4.4 or before and volumes are still attached (it needs to disable the upgrade 
 version check), the setting can not be applied to them (im-e or im-r).
- If volumes are detached, there are no im-e or im-r.